client/sysinfo/src/sysinfo.rs

```diff
@@ -534,9 +625,45 @@ pub fn gather_hwbench(scratch_directory: Option<&Path>) -> HwBench {
 			};
 	}
 
+	if is_authority {
+		ensure_requirements(hwbench.clone(), requirements);
+	}
+
 	hwbench
 }
 
+fn ensure_requirements(hwbench: HwBench, requirements: Requirements) {
+	let mut failed = 0;
+	for requirement in requirements.0.iter() {
+		match requirement.metric {
+			Metric::Blake2256 =>
+				if requirement.minimum > hwbench.cpu_hashrate_score {
+					failed += 1;
+				},
+			Metric::MemCopy =>
+				if requirement.minimum > hwbench.memory_memcpy_score {
+					failed += 1;
+				},
+			Metric::DiskSeqWrite =>
+				if let Some(score) = hwbench.disk_sequential_write_score {
+					if requirement.minimum > score {
+						failed += 1;
+					}
+				},
+			Metric::DiskRndWrite =>
+				if let Some(score) = hwbench.disk_random_write_score {
+					if requirement.minimum > score {
+						failed += 1;
+					}
+				},
+			Metric::Sr25519Verify => {},
+		}
+	}
+	if failed != 0 {
+		log::warn!("⚠️  Your hardware performance score was less than expected for role 'Authority'. See https://wiki.polkadot.network/docs/maintain-guides-how-to-validate-polkadot#reference-hardware");
+	}
```

Comment:
Why was this approved @ggwpez? Substrate != Polkadot! Please remove this link or the message entirely.

Reply:
I thought it would make it easier for the user to understand the error, but it should probably be part of the `reference_hardware.json` and not hard-coded. Will change it.

Reply:
I mean the general idea of printing the url is fine. It should just not be done by Substrate with a Polkadot url :P

Reply:
Yea I will move the printing into the CLI side of the code.